bertopic/_bertopic.py

```diff
@@ -4108,7 +4108,10 @@ def _create_topic_vectors(
             topic_embeddings = []
             topics = documents.sort_values("Topic").Topic.unique()
             for topic in topics:
-                indices = documents.loc[documents.Topic == topic, "ID"].values
+                if self._is_zeroshot():
+                    indices = documents.loc[documents.Topic == topic, "Old_ID"].values
+                else:
+                    indices = documents.loc[documents.Topic == topic, "ID"].values
```

Comment:
I would suggest that instead of this check, we just correct `ID` when combining zeroshot topics after https://github.com/MaartenGr/BERTopic/blob/fa65bc4f92f76308ec116c55dafa8c9c661ab446/bertopic/_bertopic.py#L3909
I think `documents.ID = documents.Old_ID` would work. Any reason you can think of why this wouldn't be correct?

Reply:
I tried it before here:

https://github.com/MaartenGr/BERTopic/blob/2353f4c21d74e33e34e30dbae938304bff094792/bertopic/_bertopic.py#L3847

But it didn't work there. Let me try your suggestion there and see if it helps.